qiskit/quantum_info/states/statevector.py

```diff
@@ -113,6 +113,72 @@ def __init__(
                 raise QiskitError("Invalid input: not a vector or column-vector.")
         super().__init__(op_shape=OpShape.auto(shape=shape, dims_l=dims, num_qubits_r=0))
 
+    @classmethod
+    def from_circuit(cls, circuit, ignore_set_layout=False):
```

Comment:
Could you add inline type hints here and remove them from the docstring? The suggestion here is probably too long for the line to apply it directly
```suggestion
    def from_circuit(cls, circuit: QuantumCircuit, ignore_set_layout: bool = False) -> Statevector:
```